src/pages/Voting/Voting.tsx

```diff
@@ -1,19 +1,73 @@
-import React from 'react';
+import React, {useCallback, useEffect, useState} from 'react';
 import MainLayout from '../../layouts/MainLayout/MainLayout';
 import Navigation from '../../components/Navigation/Navigation';
 import PageLayout from '../../layouts/PageLayout/PageLayout';
 import PageHeader from '../../components/PageHeader/PageHeader';
+import VotingImage from '../../components/VotingImage/VotingImage';
+import VotingActionList from '../../components/VotingActionList/VotingActionList';
+import {getFavorites, getImageForVoting, getVotes} from '../../api/requests';
+import {Favorites, Picture, Votes} from '../../types/types';
+import VotingControls from '../../components/VotingControls/VotingControls';
 
 import styles from './Voting.module.scss';
 
 function Voting() {
+  const [picture, setPicture] = useState<Picture>({} as Picture);
+  const [isLoaded, setLoad] = useState(false);
+  const [votes, setVotes] = useState<Votes[]>([]);
+  const [favorites, setFavorites] = useState<Favorites[]>([]);
+
+  const fetchVotes = useCallback(async () => {
+    const data = await getVotes();
+    setVotes(data);
+  }, []);
+
+  const fetchFavorites = useCallback(async () => {
+    const data = await getFavorites();
+    setFavorites(data);
+  }, []);
+
+  const fetchPhoto = useCallback(async () => {
+    const data = await getImageForVoting();
+    setPicture(data[0]);
+    setLoad(true);
+  }, []);
+
+  useEffect(() => {
+    fetchPhoto();
+  }, []);
+
+  useEffect(() => {
+    fetchVotes();
+  }, []);
+
+  useEffect(() => {
+    fetchFavorites();
+  }, []);
+
+  const handleButtonClick = () => {
+    fetchVotes();
+    fetchPhoto();
+    fetchFavorites();
+  };
+
+  const userActionData = [...votes ,...favorites];
```

Comment:
Лучше сортировать не в рендере VotingActionList а здесь и завернуть это в useMemo чтобы не получать новую ссылку на массив каждый раз при рендере страницы
```suggestion
  const userActionData = useMemo(() => [...votes ,...favorites].sort..........., [votes, favorites]);
```